ui/src/main/resources/FileManagerCode/RecentFilesMacro.xml

```diff
@@ -293,6 +293,19 @@
       <code>{{include reference="FileManagerCode.Macros" /}}
 
 {{velocity}}
+## Arrange paths: keep only folders names and reverse them.
+#macro(getFileFolders $fileDoc $folders)
+  #getFilePaths($fileDoc $paths)
+  #set ($paths = $paths[0])
+  #set ($folders = [])
+  #set ($size = $mathtool.sub($paths.size(), 1))
+  #foreach($i in [$size..0])
```

Comment:
Looks a bit overly complicated. Iterating in the normal order and then reversing at the end would be easier to read.